src/ConnectedMode/Binding/IBindingProcessFactory.cs

```diff
@@ -40,57 +39,31 @@ internal interface IBindingProcessFactory
     internal class BindingProcessFactory : IBindingProcessFactory
     {
         private readonly ISonarQubeService sonarQubeService;
-        private readonly IConfigurationPersister configurationPersister;
         private readonly IExclusionSettingsStorage exclusionSettingsStorage;
+        private readonly IQualityProfileDownloader qualityProfileDownloader;
         private readonly ILogger logger;
 
         [ImportingConstructor]
         public BindingProcessFactory(
             ISonarQubeService sonarQubeService,
             IExclusionSettingsStorage exclusionSettingsStorage,
-            IConfigurationPersister configurationPersister,
+            IQualityProfileDownloader qualityProfileDownloader,
             ILogger logger)
         {
             this.sonarQubeService = sonarQubeService;
-            this.configurationPersister = configurationPersister;
             this.exclusionSettingsStorage = exclusionSettingsStorage;
+            this.qualityProfileDownloader = qualityProfileDownloader;
             this.logger = logger;
         }
 
         public IBindingProcess Create(BindCommandArgs bindingArgs, bool isFirstBinding)
         {
-            var bindingConfigProvider = CreateBindingConfigProvider();
-            var solutionBindingOp = new SolutionBindingOperation();
-
-            var qpUpdater = new QualityProfileDownloader(sonarQubeService,
-                bindingConfigProvider,
-                configurationPersister,
-                solutionBindingOp,
-                logger);
-
             return new BindingProcessImpl(bindingArgs,
                 exclusionSettingsStorage,
                 sonarQubeService,
-                qpUpdater,
+                qualityProfileDownloader,
```

Comment:
The factory is now much simpler; it just imports the `IQualityProfileDownloader` and uses it. It doesn't need to create any additional objects.